### ISSUE TITLE
"./bitwarden.sh update" restarts services even if there is no update

### ISSUE THREAD

**Issue by Caligatio** [bitwarden/server#740]
Is there a reason the entire Bitwarden stack is torn down when executing the `./bitwarden.sh update` command even if there are no updates available?  It takes about 5-6 minutes for Bitwarden to fully reinitialize and `dotnet` loads several cores at 100% during this time.

**Comment by kspearrin** [comment]
We'd need to build some logic in there to detect if an update was available or not. We just pull from docker from.

**Comment by Caligatio** [comment]
The control flow for `update` in `bitwarden.sh` is a little hard to follow, but if I understand it correctly, it looks like:

- The current version of Bitwarden is baked into `bitwarden.sh` as `COREVERSION`
- On `update`, `bitwarden.sh` passes `COREVERSION` to `run.sh` which then stores it locally as `COREVERSION`
- `run.sh` then issues a "blind" `docker-compose down`
- `run.sh` then issues a `docker run` that performs an implicit pull against `bitwarden/setup:$COREVERSION`
- Presumably `bitwarden/setup:$COREVERSION` writes out the docker-compose.yml that itself has baked in version numbers on every image.
- `run.sh` then performs a docker-compose pull/up/etc

If the Docker image tags are guaranteed to align with `COREVERSION`, could you just query the Docker daemon to determine if a known container's image tag matches `COREVERSION`?  On Linux, this is something as easy as

```console
docker inspect --format='{{.Config.Image}}' bitwarden-admin | grep $COREVERSION
```

If the container isn't running, grep returns 1.  If the container is running and is current, grep returns 0.  If the container is running and is out-of-date, grep returns 1.  Basically, grep returning 1 means do your update (you might unnecessarily perform an update if the containers are down but at that point there was no teardown that I'm complaining about).

If you want to be a purist and assume nothing about running container names, you can issue a `docker-compose ps` and dynamically find a running container to do the above `docker inspect` against.

**Comment by kspearrin** [comment]
> If the Docker image tags are guaranteed to align with COREVERSION, could you just query the Docker daemon to determine if a known container's image tag matches COREVERSION?

This works for numbered versions, but we also have `latest`, `dev`, and `beta`. We'd have to use something else.

**Comment by Mart124** [comment]
See #758 👍
